pkg/image-manager/image_builder.go

```diff
@@ -312,25 +317,30 @@ func (b *ImageBuilder) imageCreate(ctx context.Context, image *Image, baseImage
 
 	tmpDir, err := ioutil.TempDir("", "image-build")
 	if err != nil {
+		log.Errorf("failed to create temp directory for building image %s: %s", image.Name, err)
 		return errors.Wrap(err, "failed to create a temp dir")
 	}
 	defer os.RemoveAll(tmpDir)
 
 	if err := images.DockerError(b.dockerClient.ImagePull(context.Background(), baseImage.DockerURL, dockerTypes.ImagePullOptions{})); err != nil {
+		log.Errorf("failed to pull base image %s: %s", baseImage.DockerURL, err)
 		return errors.Wrapf(err, "failed to pull image '%s'", baseImage.DockerURL)
 	}
 
 	if err := b.copyImageTemplate(tmpDir, baseImage.DockerURL); err != nil {
+		log.Errorf("failed to copy image template from image %s: %s", image.Name, err)
 		return err
 	}
 
 	spFile := filepath.Join(tmpDir, systemPackagesFile)
 	if err := b.writeSystemPackagesFile(spFile, image); err != nil {
+		log.Errorf("failed to write packages file for image %s [%s]: %s", image.Name, spFile, err)
```

Comment:
I feel like we need a `LogAndWrap(err error, message string, format... interface{}) error` function :-)